src/memphis.ts

```diff
@@ -376,22 +453,47 @@ class Producer {
         asyncProduce = false,
         headers = new MsgHeaders()
     }: {
-        message: Uint8Array;
+        message: any;
         ackWaitSec?: number;
         asyncProduce?: boolean;
         headers?: MsgHeaders;
     }): Promise<void> {
         try {
+            let messageToSend;
             headers.headers.set('$memphis_connectionId', this.connection.connectionId);
             headers.headers.set('$memphis_producedBy', this.producerName);
-            const subject = this.stationName.replace(/\./g, '#');
+            let meassageDescriptor = this.connection.meassageDescriptors.get(this.internal_station);
+            if (meassageDescriptor) {
+                if (message instanceof Uint8Array) {
+                    messageToSend = message;
+                    meassageDescriptor.decode(messageToSend);
+                } else if (message instanceof Object) {
+                    let errMsg = meassageDescriptor.verify(message);
+                    if (errMsg) {
+                        throw Error(errMsg);
+                    }
+                    messageToSend = meassageDescriptor.encode(message).finish();
+                } else {
+                    throw Error('Unsupported message type');
+                }
+            }
             if (asyncProduce)
-                this.connection.brokerConnection.publish(`${subject}.final`, message, { headers: headers.headers, ackWait: ackWaitSec * 1000 * 1000000 });
-            else await this.connection.brokerConnection.publish(`${subject}.final`, message, { headers: headers.headers, ackWait: ackWaitSec * 1000 * 1000000 });
+                this.connection.brokerConnection.publish(`${this.internal_station}.final`, messageToSend, {
+                    headers: headers.headers,
+                    ackWait: ackWaitSec * 1000 * 1000000
+                });
+            else
+                await this.connection.brokerConnection.publish(`${this.internal_station}.final`, messageToSend, {
+                    headers: headers.headers,
+                    ackWait: ackWaitSec * 1000 * 1000000
+                });
         } catch (ex: any) {
             if (ex.code === '503') {
                 throw new Error('Produce operation has failed, please check whether Station/Producer are still exist');
             }
+            if (ex.message?.includes('missing required') || ex.message?.includes('expected')) {
```

Comment:
this is not the way to catch schema error.
In the section below throw an error as I wrote yesterday, this exception will get into the catch block and will be thrown to the user